apps/condo/domains/common/utils/next/helpers.ts

```diff
@@ -0,0 +1,20 @@
+import { NextPageContext } from 'next'
+import Router from 'next/router'
+
+import { isSSR } from '@open-condo/miniapp-utils'
+
+
+type NextRedirectParams = { destination: string, permanent?: boolean, as?: string }
+type NextRedirectReturnType = { pageProps: Record<string,  never> }
+export async function nextRedirect (pageContext: NextPageContext, params: NextRedirectParams): Promise<NextRedirectReturnType> {
+    if (isSSR()) {
+        pageContext.res.writeHead(params.permanent ? 308 : 307, {
+            Location: params.destination,
+        })
+        pageContext.res.end()
+    } else {
+        await Router.push(params.destination, params.as)
+    }
+
+    return { pageProps: {} }
```

Comment:
Why do you need to return props in this util?

Reply:
if you do a redirect, you should to return empty page props. I thought it would be a little more convenient. But you can always return any result